Drop IsDeprecatedWeakRefSmartPointerException for more classes

cdumez opened the pull request:
#### fb3655a31cceea420f8cf2ea19c60f8f4b4ca696
<pre>
Drop IsDeprecatedWeakRefSmartPointerException for more classes
<a href="https://bugs.webkit.org/show_bug.cgi?id=300376">https://bugs.webkit.org/show_bug.cgi?id=300376</a>

Reviewed by Darin Adler.

* Source/WebCore/Modules/speech/SpeechRecognizer.cpp:
(WebCore::SpeechRecognizer::startCapture):
* Source/WebCore/Modules/speech/SpeechRecognizer.h:
(WebCore::SpeechRecognizer::source): Deleted.
(WebCore::SpeechRecognizer::setInactive): Deleted.
* Source/WebCore/PAL/pal/system/mac/SystemSleepListenerMac.h:
* Source/WebCore/PAL/pal/system/mac/SystemSleepListenerMac.mm:
(PAL::SystemSleepListenerMac::SystemSleepListenerMac):
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::SameSizeAsScrollableArea::checkedPtrCount const):
(WebCore::SameSizeAsScrollableArea::checkedPtrCountWithoutThreadCheck const):
(WebCore::SameSizeAsScrollableArea::incrementCheckedPtrCount const):
(WebCore::SameSizeAsScrollableArea::decrementCheckedPtrCount const):
(WebCore::SameSizeAsScrollableArea::~SameSizeAsScrollableArea):
(): Deleted.
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.h:
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.mm:
(WebCore::SharedRoutingArbitrator::beginRoutingArbitrationForToken):
* Source/WebCore/platform/cf/MainThreadSharedTimerCF.cpp:
(WebCore::setupPowerObserver):
* Source/WebCore/platform/cocoa/PowerSourceNotifier.h:
* Source/WebCore/platform/cocoa/PowerSourceNotifier.mm:
(WebCore::PowerSourceNotifier::PowerSourceNotifier):
* Source/WebCore/platform/mac/PowerObserverMac.cpp:
(WebCore::PowerObserver::didReceiveSystemPowerNotification):
* Source/WebCore/platform/mac/PowerObserverMac.h:
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::convertSVGToOTFFont):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
(WebCore::BackgroundFetchResponseBodyLoader::BackgroundFetchResponseBodyLoader): Deleted.
* Source/WebKit/UIProcess/SpeechRecognitionServer.cpp:
(WebKit::SpeechRecognitionServer::checkedRecognizer const):
(WebKit::SpeechRecognitionServer::handleRequest):
(WebKit::SpeechRecognitionServer::stop):
(WebKit::SpeechRecognitionServer::abort):
(WebKit::SpeechRecognitionServer::invalidate):
* Source/WebKit/UIProcess/SpeechRecognitionServer.h:

Canonical link: <a href="https://commits.webkit.org/301240@main">https://commits.webkit.org/301240@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a719a2b1c310874df1b2c46b68c187818fc86044

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125322 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44987 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35728 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132171 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77191 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c34adbd4-b6b6-493d-9e40-e25168ba3e62) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127199 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45675 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53552 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95416 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63346 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ef761918-a0e8-45b0-a30f-54f66f6322ca) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128276 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36478 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112072 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75955 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b271f4c9-217e-4456-b328-3bf5b5e098ce) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35377 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30241 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75648 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106246 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30467 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134857 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52125 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39904 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103886 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52563 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108294 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103647 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26402 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49010 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27305 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49235 "Hash a719a2b1 for PR 51994 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52020 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57800 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51377 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54733 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53071 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/115c8ca1f51d2385b5d46044dfb1cc3b59f3105a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51994

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/61cf9a8d0f150a21091225798f7d0216c5897567. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51994

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a5a8b7d6ca56da6b314401883934768d17ddafaa. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51994

cdumez committed:
Drop IsDeprecatedWeakRefSmartPointerException for more classes
https://bugs.webkit.org/show_bug.cgi?id=300376

Reviewed by Darin Adler.

* Source/WebCore/Modules/speech/SpeechRecognizer.cpp:
(WebCore::SpeechRecognizer::startCapture):
* Source/WebCore/Modules/speech/SpeechRecognizer.h:
(WebCore::SpeechRecognizer::source): Deleted.
(WebCore::SpeechRecognizer::setInactive): Deleted.
* Source/WebCore/PAL/pal/system/mac/SystemSleepListenerMac.h:
* Source/WebCore/PAL/pal/system/mac/SystemSleepListenerMac.mm:
(PAL::SystemSleepListenerMac::SystemSleepListenerMac):
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::SameSizeAsScrollableArea::checkedPtrCount const):
(WebCore::SameSizeAsScrollableArea::checkedPtrCountWithoutThreadCheck const):
(WebCore::SameSizeAsScrollableArea::incrementCheckedPtrCount const):
(WebCore::SameSizeAsScrollableArea::decrementCheckedPtrCount const):
(WebCore::SameSizeAsScrollableArea::~SameSizeAsScrollableArea):
(): Deleted.
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.h:
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.mm:
(WebCore::SharedRoutingArbitrator::beginRoutingArbitrationForToken):
* Source/WebCore/platform/cf/MainThreadSharedTimerCF.cpp:
(WebCore::setupPowerObserver):
* Source/WebCore/platform/cocoa/PowerSourceNotifier.h:
* Source/WebCore/platform/cocoa/PowerSourceNotifier.mm:
(WebCore::PowerSourceNotifier::PowerSourceNotifier):
* Source/WebCore/platform/mac/PowerObserverMac.cpp:
(WebCore::PowerObserver::didReceiveSystemPowerNotification):
* Source/WebCore/platform/mac/PowerObserverMac.h:
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::convertSVGToOTFFont):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
(WebCore::BackgroundFetchResponseBodyLoader::BackgroundFetchResponseBodyLoader): Deleted.
* Source/WebKit/UIProcess/SpeechRecognitionServer.cpp:
(WebKit::SpeechRecognitionServer::checkedRecognizer const):
(WebKit::SpeechRecognitionServer::handleRequest):
(WebKit::SpeechRecognitionServer::stop):
(WebKit::SpeechRecognitionServer::abort):
(WebKit::SpeechRecognitionServer::invalidate):
* Source/WebKit/UIProcess/SpeechRecognitionServer.h:

Canonical link: https://commits.webkit.org/301240@main
